app/controllers/tweets_controller.rb

```diff
@@ -0,0 +1,72 @@
+require 'rack-flash'
+
+class TweetsController < ApplicationController
+
+  use Rack::Flash
+
+  get '/tweets/new' do
+    redirect to '/login' unless logged_in?
+
+    erb :'tweets/create_tweet'
+  end
+
+  get '/tweets/:id' do
+    redirect to '/login' unless logged_in?
+
+    @tweet = Tweet.find(params[:id])
+    erb :'/tweets/show_tweet'
+  end
+
+  get '/tweets/:id/edit' do
+    redirect to '/login' unless logged_in?
+
+    @tweet = Tweet.find(params[:id])
+
+    if @tweet.user.id != current_user.id
+      flash[:notice] = "You cannot edit another user's tweet!"
+      redirect to "/tweets/#{@tweet.id}"
+    end
+
+    erb :'/tweets/edit_tweet'
+  end
+
+  get '/tweets' do
+    redirect to '/login' unless logged_in?
+
+    @tweets = Tweet.all
+    erb :'/tweets/tweets'
+  end
+
+  post '/tweets' do
+    @tweet = Tweet.new(content: params[:content], user_id: current_user.id)
```

Comment:
Another way to do this is to make use of the `has_many` relationship on User:
```
@tweet = current_user.tweets.create(content: params[:content])
```
Both approaches are good, but in some cases you may find this option useful.